src/frontend/apps/desk/src/components/DropButton.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { PropsWithChildren, ReactNode, useState } from 'react';
+import { Button, DialogTrigger, Popover } from 'react-aria-components';
+import styled from 'styled-components';
+
+const StyledPopover = styled(Popover)`
+  background-color: white;
+  border-radius: 4px;
+  box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.1);
+  padding: 0.5rem;
+  border: 1px solid #dddddd;
+  opacity: 0;
+  transition: opacity 0.2s ease-in-out;
+`;
+
+const StyledButton = styled(Button)`
+  cursor: pointer;
+  border: none;
+  background: none;
+  outline: none;
+  transition: all 0.2s ease-in-out;
+`;
+
+interface DropButtonProps {
+  button: ReactNode;
+  onOpenChange?: (isOpen: boolean) => void;
+}
+
+export const DropButton = ({
+  button,
+  onOpenChange,
+  children,
+}: PropsWithChildren<DropButtonProps>) => {
+  const [opacity, setOpacity] = useState(false);
+
+  return (
+    <DialogTrigger
+      onOpenChange={(isOpen) => {
+        onOpenChange?.(isOpen);
+        setTimeout(() => {
+          setOpacity(isOpen);
+        }, 10);
+      }}
+    >
+      <StyledButton>{button}</StyledButton>
+      <StyledPopover style={{ opacity: opacity ? 1 : 0 }}>
+        {children}
+      </StyledPopover>
+    </DialogTrigger>
```

Comment:
I wonder why not stick to react-aria first example, using their `MenuTigger` and `MenuItem` components.
Do you prefer using Cunningham `Button` instead of their `MenuItem`?

(I talk about this example)
```
import {Button, Menu, MenuItem, MenuTrigger, Popover} from 'react-aria-components';

<MenuTrigger>
  <Button aria-label="Menu">☰</Button>
  <Popover>
    <Menu onAction={alert}>
      <MenuItem id="open">Open</MenuItem>
      <MenuItem id="rename">Rename…</MenuItem>
      <MenuItem id="duplicate">Duplicate</MenuItem>
      <MenuItem id="share">Share…</MenuItem>
      <MenuItem id="delete">Delete…</MenuItem>
    </Menu>
  </Popover>
</MenuTrigger>
```